### PR TITLE
Added test workflow with basic python test

### PR DESCRIPTION
**Pull request by lynnro314**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: iceberg-tests
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  iceberg-isolation-test:
+    name: Test lakeFS isolation with iceberg
```
Comment (johnnyaug) [inline .github/workflows/tests.yml:10]: Why specifically "isolation"? This is testing lakeFS with Iceberg

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    runs-on: ubuntu-20.04
+    services:
+      lakefs:
+        image: "treeverse/lakefs:latest"
```
Comment (johnnyaug) [inline .github/workflows/tests.yml:14]: May be better to lock the version number here. There is a trade-off, but I think consistency wins over freshness.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+      - name: Generate uniquifying value
+        id: unique
+        run: echo "value=$RANDOM" >> $GITHUB_OUTPUT
+
+      - name: Use lakeFS with S3 gateway
+        run: |
+          python ./test/run-test.py \
+            --storage_namespace s3://iceberg-lakefs-testing/${{ github.run_number }}-s3-gateway/${{ steps.unique.outputs.value }} \
+            --repository gateway-test \
```
Comment (johnnyaug) [inline .github/workflows/tests.yml:54]: newline

File: test/run-test.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    df = spark.read.option("inferSchema","true").option("multiline","true").csv("./test/data-sets/film_permits.csv")
+    df.write.saveAsTable("lakefs.main.nyc.permits")
+
+    #Commit, create a new branch, check that the tables are the same
+    lfs_client.commits.commit(repo_name, "main", CommitCreation(message="Initial data load"))
+    lfs_client.branches.create_branch(repo_name, BranchCreation(name="dev", source="main"))
+    df_main = spark.read.table("lakefs.main.nyc.permits")
+    df_dev = spark.read.table("lakefs.dev.nyc.permits")
+    assert (df_main.schema == df_dev.schema) and (df_main.collect() == df_dev.collect()), "main and dev tables should be equal"
```
Comment (johnnyaug) [inline test/run-test.py:63]: In a future PR, can we use some Python testing framework here? To manage the test cases and report the failures.

**Review by johnnyaug — APPROVED** [review]
Nice! minor comments

File: test/run-test.py (hunk 4 above)
Comment (lynnro314) [reply test/run-test.py:63]: Of course, this is how it will be.

**Pull request merged**